src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -386,7 +381,16 @@ public virtual void Challenge(IChallengeContext context)
         /// <returns></returns>
         protected virtual Task ApplyResponseChallengeAsync()
         {
-            ApplyResponseChallenge();
+            // If authenticate was called and the the status is still 401, authZ failed so set 403 and stop
+            // REVIEW: Does this need to ensure that there's the 401 is challenge for this auth type?
+            if (ShouldConvertChallengeToForbidden())
```

Comment:
Actually now that I remember some of our earlier conversations, this was supposed to move back down into CookieAuth & BearerAuth's ApplyResponseChallenge because CookieAuth doesn't want to issue 403s, it wants to redirect to an AccessDenied page similar to how it handles challenges.